ufo2fontir/src/source.rs

```diff
@@ -478,35 +474,196 @@ fn files_identical(f1: &Path, f2: &Path) -> Result<bool, WorkError> {
+        builder.add_if_present(NameId::Trademark, &font_info.trademark);
+        builder.add_if_present(
+            NameId::ManufacturerName,
+            &font_info.open_type_name_manufacturer,
+        );
+        builder.add_if_present(NameId::Designer, &font_info.open_type_name_designer);
+        builder.add_if_present(NameId::Description, &font_info.open_type_name_description);
+        builder.add_if_present(
+            NameId::ManufacturerUrl,
+            &font_info.open_type_name_manufacturer_url,
+        );
+        builder.add_if_present(NameId::DesignerUrl, &font_info.open_type_name_designer_url);
+        builder.add_if_present(NameId::License, &font_info.open_type_name_license);
+        builder.add_if_present(NameId::LicenseUrl, &font_info.open_type_name_license_url);
+        builder.add_if_present(
+            NameId::MacCompatibleFullName,
+            &font_info.open_type_name_compatible_full_name,
+        );
+        builder.add_if_present(NameId::SampleText, &font_info.open_type_name_sample_text);
+        builder.add_if_present(
+            NameId::WwsFamilyName,
+            &font_info.open_type_name_wws_family_name,
+        );
+        builder.add_if_present(
+            NameId::WwsSubfamilyName,
+            &font_info.open_type_name_wws_subfamily_name,
+        );
+
+        // After our first pass at getting values, apply fallbacks
+
+        // https://github.com/googlefonts/ufo2ft/blob/fca66fe3ea1ea88ffb36f8264b21ce042d3afd05/Lib/ufo2ft/fontInfoData.py#L188
+        builder.apply_fallback(NameId::TypographicFamilyName, &[NameId::FamilyName]);
+
+        // https://github.com/googlefonts/ufo2ft/blob/fca66fe3ea1ea88ffb36f8264b21ce042d3afd05/Lib/ufo2ft/fontInfoData.py#L195
+        builder.apply_fallback(NameId::TypographicSubfamilyName, &[NameId::SubfamilyName]);
+
+        // Version has a weird fallback
+        // https://github.com/googlefonts/ufo2ft/blob/fca66fe3ea1ea88ffb36f8264b21ce042d3afd05/Lib/ufo2ft/fontInfoData.py#L169
+        if !builder.contains_key(NameId::Version) {
+            let major = font_info.version_major.unwrap_or_default();
+            let minor = font_info.version_minor.unwrap_or_default();
+            builder.add(NameId::Version, format!("Version {major}.{minor:0>3}"));
+        }
+
+        // Full name is always based on typographi family
+        builder.add(
+            NameId::FullName,
+            format!(
+                "{} {}",
+                builder
+                    .get(NameId::TypographicFamilyName)
+                    .unwrap_or_default(),
+                builder
+                    .get(NameId::TypographicSubfamilyName)
+                    .unwrap_or_default(),
+            ),
+        );
+
+        // Name's that don't get individual fields
+        if let Some(name_records) = font_info.open_type_name_records.as_ref() {
+            for nr in name_records.iter() {
+                let name_id: u16 = nr.name_id.try_into().unwrap();
```

Comment:
UFO's openTypeNameRecords not only specify the nameID, but also the specific platformID, platEncID and langID; you seem to be only using the nameID here and ignoring the rest.

See https://unifiedfontobject.org/versions/ufo3/fontinfo.plist/#name-record-format

Reply:
I suppose you are intentionally ignoring the additional attributes because of the complications of encoding the string, if that's the case I suggest you add a TODO as reminder

Reply:
https://github.com/googlefonts/fontmake-rs/issues/188